modules/nf-core/quilt/quilt/main.nf

```diff
@@ -0,0 +1,62 @@
+process QUILT_QUILT {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "bioconda::r-quilt=1.0.5"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/r-quilt:1.0.5--r43h06b5641_0':
+        'biocontainers/r-quilt:1.0.5--r43h06b5641_0' }"
+
+    input:
+    tuple val(meta), path(bams), path(bais), path(bamlist), val(chr), val(regions_start), val(regions_end), val(buffer), val(ngen), path(reference_haplotype_file), path(reference_legend_file), path(genetic_map_file)
```

Comment:
That is a really big tuple...
Is the order specific to anything? Can the mandatory file paths come before the optional arguments?
What are the value channels doing?

Reply:
Yes, the tuple contains a substantial number of arguments, which we see is a common requirement for imputation tools. A similar structure can be found in tools such as [STITCH](https://github.com/nf-core/modules/blob/master/modules/nf-core/stitch/main.nf) and [GLIMPSE 1](https://github.com/nf-core/modules/blob/master/modules/nf-core/glimpse/phase/main.nf) and [2](https://github.com/nf-core/modules/blob/master/modules/nf-core/glimpse2/phase/main.nf).

The value channels within the tuple serve the purpose of defining essential variable information, including the chromosome and region to be processed. While nGen, seed, and buffer are currently included in the tuple, I appreciate your suggestion to potentially provide them as ext.args. I will work on implementing this improvement.

Reply:
Ok, so if they are based on a previous tool output then a value makes sense.

Reply:
Just wanted to add here that nGen is a mandatory parameter for the tool and per nf-core guidelines it should be supplied as a value channel and not in ext.args 😄

Reply:
Hmm, ok, if they are not set to any kind of default then they should be value channels: https://nfcore.slack.com/archives/C043UU89KKQ/p1677581560661679
What else is required for the tool to run and has no default set at all?

Reply:
`nGen` default is NA
`buffer` default is NA
`seed` default is also NA (but we added a default)

Reply:
@atrigila  can quilt run without specifying those parameters? I know that STITCH requires nGen so I supposed that also QUILT does but probably you know better than me? :) I think buffer is probably optional?

Reply:
I've tested them and unfortunately they are not optional, the code breaks. So I added them again back as val channels.

Reply:
Thank you both for your reviews! @SPPearce @saulpierotti